src/cli/syncback.rs

```diff
@@ -0,0 +1,292 @@
+                .insert(*name, property.to_owned());
+        }
+
+        let children = dom.clone_multiple_into_external(real_root.children(), &mut new_tree);
+        for child in children {
+            new_tree.transfer_within(child, new_tree.root_ref());
+        }
+        Ok(new_tree)
+    } else {
+        anyhow::bail!(
+            "Rojo does not currently support models with more \
+        than one Instance at the Root!"
+        );
+    }
+}
+
+fn xml_decode_config() -> rbx_xml::DecodeOptions<'static> {
+    rbx_xml::DecodeOptions::new().property_behavior(rbx_xml::DecodePropertyBehavior::ReadUnknown)
+}
+
+/// The different kinds of input that Rojo can syncback.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum FileKind {
+    /// An XML model file.
+    Rbxmx,
+
+    /// An XML place file.
+    Rbxlx,
+
+    /// A binary model file.
+    Rbxm,
+
+    /// A binary place file.
+    Rbxl,
+}
+
+impl FileKind {
+    fn from_path(output: &Path) -> Option<FileKind> {
+        let extension = output.extension()?.to_str()?;
+
+        match extension {
+            "rbxlx" => Some(FileKind::Rbxlx),
+            "rbxmx" => Some(FileKind::Rbxmx),
+            "rbxl" => Some(FileKind::Rbxl),
+            "rbxm" => Some(FileKind::Rbxm),
+            _ => None,
+        }
+    }
+}
+
+fn list_files(snapshot: &FsSnapshot, color: ColorChoice, base_path: &Path) -> io::Result<()> {
+    let no_color = ColorSpec::new();
+    let mut add_color = ColorSpec::new();
+    add_color.set_fg(Some(Color::Green));
+    let mut remove_color = ColorSpec::new();
+    remove_color.set_fg(Some(Color::Red));
+
+    // We emit this to stderr because otherwise it'd be impossible
+    // to pipe it separately from normal output.
+    let writer = BufferWriter::stderr(color);
+    let mut buffer = writer.buffer();
+
+    if snapshot.is_empty() {
+        writeln!(
+            &mut buffer,
+            "No files/directories would be removed or added."
+        )?;
+    } else {
+        let added = snapshot.added_paths();
+        if !added.is_empty() {
+            writeln!(&mut buffer, "Writing files/directories:")?;
+            buffer.set_color(&add_color)?;
+            for path in added {
+                writeln!(
+                    &mut buffer,
+                    "{}",
+                    display_absolute(path.strip_prefix(base_path).unwrap_or(path))
+                )?;
+            }
+            buffer.set_color(&no_color)?;
+        }
+        let removed = snapshot.removed_paths();
+        if !removed.is_empty() {
+            writeln!(&mut buffer, "Removing files/directories:")?;
+            buffer.set_color(&remove_color)?;
+            for path in removed {
+                writeln!(
+                    &mut buffer,
+                    "{}",
+                    display_absolute(path.strip_prefix(base_path).unwrap_or(path))
+                )?;
+            }
+        }
+        buffer.set_color(&no_color)?;
+    }
```

Comment:
A user running syncback in interactive mode will already know that zero files would be touched from earlier messages, and programmatic usage is simpler if stdout stays empty when there are zero writes.

We can skip checking whether the entire snapshot is empty and just check additions and removals. Along with the output format changes, that'd be:
```suggestion
    let added = snapshot.added_paths();
    if !added.is_empty() {
        buffer.set_color(&add_color)?;
        for path in added {
            writeln!(
                &mut buffer,
                "Writing {}",
                display_absolute(path.strip_prefix(base_path).unwrap_or(path))
            )?;
        }
    }
    let removed = snapshot.removed_paths();
    if !removed.is_empty() {
        buffer.set_color(&remove_color)?;
        for path in removed {
            writeln!(
                &mut buffer,
                "Removing {}",
                display_absolute(path.strip_prefix(base_path).unwrap_or(path))
            )?;
        }
    }
    buffer.set_color(&no_color)?;
```